cmd/golangBuild.go

```diff
@@ -404,6 +427,51 @@ func reportGolangTestCoverage(config *golangBuildOptions, utils golangBuildUtils
 	return nil
 }
 
+func retrieveGolangciLint(utils golangBuildUtils, golangciLintDir string) error {
+	// installation instructions: https://golangci-lint.run/usage/install/#linux-and-windows
+	var outputBuffer bytes.Buffer
+	utils.Stdout(&outputBuffer)
+	err := utils.RunExecutable("curl", "-sSfL", golangciLintCurlURL)
```

Comment:
agree, download should not make assumption about installation of `curl`

Reply:
```retrieveGolangciLint``` function was deleted. 
Added usage of ```go install``` instead of ```curl``` - [829b6c2](https://github.com/SAP/jenkins-library/pull/3903/commits/829b6c2426fc534021b8c9e2fb8b536b385f28f0)

Reply:
Added usage of ```http``` package

Reply:
Added usage of ```DownloadFile```